cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java

```diff
@@ -147,6 +149,18 @@ public void augmentSubject(final Subject subject)
             PluginFactory pluginFactory = new PluginFactory();
             UserPersistence userPersistence = pluginFactory.createUserPersistence();
             Principal ldapPrincipal = getLdapPrincipal(subject);
+
+            // Remove the HttpPrincipal in subject, if any
+            Set<Principal> toBeRemoved = new HashSet<Principal>();
+            subject.getPrincipals().forEach(p -> {
+                if (p instanceof HttpPrincipal) {
+                    toBeRemoved.add(p);
+                }
+            });
+            toBeRemoved.forEach(p -> {
+                subject.getPrincipals().remove(p);
+            });
+
```

Comment:
Would `removeAll(toBeRemoved)` be easier to read here?

Reply:
Good idea. This does work.